otaclient/ota_proxy/utils.py

```diff
@@ -39,3 +45,32 @@ async def hexdigest(self) -> str:
         return await asyncio.get_running_loop().run_in_executor(
             self._executor, self._hashf.hexdigest
         )
+
+
+class TmpCacheFileNaming:
```

Comment:
New naming scheme is introduced to naming the tmp ongoing caching file predictably with the requested URL.
This makes ota_proxy aware of the relationship between a tmp file and the URL it caches for.

This is intended for the future feature of resume downloading OTA file.